### PR TITLE
[#139179971] Upgrade CF to v253

### PR DESCRIPTION
**Pull request by keymon**
[#139179971 Upgrade CF to >= v252](https://www.pivotaltracker.com/story/show/139179971)

What?
-----

We want to upgrade CF to v253.

In this PR we do so. Changes to highlight:

 * Added mutual TLS for cloudcontroller, tps, etc.
 * Prepare manifest to use bosh links in the future.
 * Upgraded to the latest stemcell in the recommended series.

We had to patch the cf-acceptance-tests in https://github.com/cloudfoundry/cf-acceptance-tests/pull/188 and we added a hardcoded checkout in the task to use our patched version.

Dependencies
--------------------

 * PR for the docker image to update the cf-client https://github.com/alphagov/paas-docker-cloudfoundry-tools/pull/91 Once merged, remove the temporary commit.

How to test?
-----------

 - Code review.
 - Review notes from pivotal story.
 - Deployment test:
   1. deploy from master
   2. deploy from this branch
   3. Check all tests pass

Who?
----

Anyone but @keymon or @csaliceti.

### CODE REVIEW

**Comment by henrytk** [comment]
@keymon @saliceti - the build is currently failing due to Shellcheck errors.

**Comment by keymon** [comment]
Yes, sorry, I forgot to push the fix. I removed the TEMP commit and rebased with master too.

File: manifests/cf-manifest/manifest/010-cf-jobs.yml [hunk 1]
```diff
@@ -8,8 +8,10 @@ meta:
   api_templates:
   - name: consul_agent
     release: (( grab meta.consul_templates.consul_agent.release ))
+    consumes: {consul: nil}
```
Comment (henrytk) [inline manifests/cf-manifest/manifest/010-cf-jobs.yml]: Out of curiousity, do these Bosh links provide a native solution to the problem that resulted in our creation of https://github.com/alphagov/paas-datadog-for-cloudfoundry-boshrelease ? The problem, if I remember correctly, was the datadog agent trying to access properties from other jobs.

Comment (keymon) [inline manifests/cf-manifest/manifest/010-cf-jobs.yml]: Not sure, but they provide a scoped way to share connection details between "jobs" in a deployed. For instance, you can deploy multiple different consul clusters, and then consume the details from one to the other. 

It also prevents having to repeat the properties of one server in the clients.

File: concourse/pipelines/create-cloudfoundry.yml [hunk 2]
```diff
@@ -1942,6 +1942,23 @@ jobs:
                 - -e
                 - -c
                 - |
+                  # FIXME: Remove this once
+                  # https://github.com/cloudfoundry/cf-acceptance-tests/pull/188 has been merged
```
Comment (henrytk) [inline concourse/pipelines/create-cloudfoundry.yml]: If there is a story for the next upgrade already created we should add a comment about this.

Comment (keymon) [inline concourse/pipelines/create-cloudfoundry.yml]: There is not, but I can add it. Anyway, that code will fail if you change the version of cf-release

**Comment by dcarley** [comment]
Should we include the revert of e5c79dd4e93286a7340b6ee5af7042508630a0ee in this PR?

**Comment by keymon** [comment]
> Should we include the revert of e5c79dd in this PR?

Yes! I will

**Pull request merged**